homeassistant/components/mcp23017/binary_sensor.py

```diff
@@ -0,0 +1,87 @@
+"""Support for binary sensor using I2C MCP23017 chip."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import DEVICE_DEFAULT_NAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_INVERT_LOGIC = 'invert_logic'
+CONF_I2C_ADDRESS = 'i2c_address'
+CONF_PINS = 'pins'
+CONF_PULL_MODE = 'pull_mode'
+
+DEFAULT_INVERT_LOGIC = False
+DEFAULT_I2C_ADDRESS = 0x20
+DEFAULT_PULL_MODE = 'UP'
+
+_SENSORS_SCHEMA = vol.Schema({
+    cv.positive_int: cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PINS): _SENSORS_SCHEMA,
+    vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+    vol.Optional(CONF_PULL_MODE, default=DEFAULT_PULL_MODE): vol.In(['UP',
+                                                                     'DOWN']),
+    vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS):
+    vol.Coerce(int),
+})
+
+
+async def async_setup_platform(hass, config, async_add_devices,
```

Comment:
Let's make it `setup_platform` if you agree that the setup could also do I/O.

Reply:
You are right, it could also do I/O. let's do that.